GLMakie/src/drawing_primitives.jl

```diff
@@ -767,18 +772,20 @@ function draw_atomic(screen::Screen, scene::Scene, plot::Surface)
 
         if all(T -> T <: Union{AbstractMatrix, AbstractVector}, types)
             t = Makie.transform_func_obs(plot)
+            f32c = scene.float32convert
+            f32c_obs = f32c isa Makie.Float32Convert ? f32c.scaling : Observable(nothing)
             mat = plot[3]
-            xypos = lift(plot, t, plot[1], plot[2], space) do t, x, y, space
+            xypos = lift(plot, f32c_obs, t, plot[1], plot[2], space) do f32c_inner, t, x, y, space
                 # Only if transform doesn't do anything, we can stay linear in 1/2D
-                if Makie.is_identity_transform(t)
+                if Makie.is_identity_transform(t) && isnothing(f32c_inner)
                     return (x, y)
                 else
                     matrix = if x isa AbstractMatrix && y isa AbstractMatrix
-                        apply_transform.((t,), Point.(x, y), space)
+                        Makie.f32_convert(f32c, apply_transform.((t,), Point.(x, y), space), space)
                     else
                         # If we do any transformation, we have to assume things aren't on the grid anymore
                         # so x + y need to become matrices.
-                        [apply_transform(t, Point(x, y), space) for x in x, y in y]
+                        [MAkie.f32_convert(f32c, apply_transform(t, Point(x, y), space), space) for x in x, y in y]
```

Comment:
```suggestion
                        [Makie.f32_convert(f32c, apply_transform(t, Point(x, y), space), space) for x in x, y in y]
```
(typo fix)

Reply:
Thanks, but I don't think it's worth reviewing details at this point. I'm currently prototyping without much care and plan to clean things up later, once we are more certain about whether this is a good direction to go.

Reviews on a more conceptual level would be more useful. For example if you see issues with how this would integrate with GeoMakie. I think I noted this somewhere before but the rough plan for the conversion pipeline here is:
1. convert_arguments makes structural changes, e.g. Vector, Vector -> Vector{Point2}, leaving types alone or converting to Float32 and Float64 as appropriate
2. apply transform_func, preferably keeping types
3. apply Float32 conversion which may rescale data
4. continue with model, camera as usual

I think Axis converts apply between 1 and 2, but I haven't paid much attention to that pr yet, tbh.

Also regarding transformed space - it's not the point of this pr to add it but I wouldn't mind. I want to add that and world space, but it's not a priority for me.

Reply:
>I think Axis converts apply between 1 and 2, but I haven't paid much attention to that pr yet, tbh.

No, axis convert shouldn't change anything about that at this point.